Refactor current query terms

VasilyShelkov opened the pull request:
-Changed #searchKeyword css to #addSearchTerm to be consistent with name of component
-Extracted AddSearchTerm into its own component
-Using ids with jquery instead of react refs since it was the way to achieve the onclick and focus in the least amount of code and also so selecting the container and addsearchterm is consistently using ids instead of one using refs and the other using id selector 

@Winwardo ready to be checked and merged if you're happy with it.

VasilyShelkov committed:
Started refactoring but have train to catch !

VasilyShelkov committed:
Merged branch master into refactor-current-query-terms

VasilyShelkov committed:
Refactor 1st draft complete

Winwardo commented on src/client/search/AddSearchTerm.js:
```diff
@@ -0,0 +1,36 @@
+    borderTop: '2px dashed #D3D5D8',
+  };
+  return (
+    <div className="ui fluid big transparent input"
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2226118788/issues/source?bid=3148782&fileBranchId=3174638#l15)

Winwardo commented on src/client/search/AddSearchTerm.js:
```diff
@@ -0,0 +1,36 @@
+  return (
+    <div className="ui fluid big transparent input"
+      style={keywordSearchStyle}
+      id="searchTermContainer">
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2226118788/issues/source?bid=3148782&fileBranchId=3174638#l17)

Winwardo commented on src/client/search/CurrentQueryTerms.js:
```diff
@@ -1,56 +1,30 @@
+const CurrentQueryTerms = () => {
+  const showAddSearchTerm = () => {
+    //searchTermContainer and addSearchTerm are in ./AddSearchTerm
+    if ($('#searchTermContainer').is(':hidden')) {
```
Please add to the comment that you're aware this is a bad practise, but is done for the sake for simplicity and ease of coding

Winwardo commented on src/client/search/CurrentQueryTerms.js:
```diff
@@ -1,56 +1,30 @@
-						}}/>
-          <i className="link remove circle icon"></i>
-        </div>
+        <AddSearchTerm addSerchTermRef={node => searchTermRef = node}/>
```
*addSearchTermRef, not addSerchTermRef

Winwardo commented:
All seems good :+1:

VasilyShelkov committed:
Fixed Pull Request according to comment

-Put a more informative comment
-Got rid of the ref entirely
-Added eslint env props to acknowledge use of Jquery and Mocha

VasilyShelkov commented:
@Winwardo Ready to push now that fixed your comments :)
